search for HDF5 header even when nc-config used in setup.py

jswhit opened the pull request:
and use H5lib_getversion to get version info (instead of getting it from
header).

thehesiod commented on setup.py:
```diff
@@ -2,9 +2,10 @@
-    "install_requires":  ["numpy>=1.7"],
-    "setup_requires":  ['setuptools>=18.0',"cython>=0.19"],
+    "install_requires": ["numpy>=1.7"],
+    "setup_requires": ['setuptools>=18.0', "cython>=0.19", ""],
```
hmm, seems like pycharm code reformat added an empty import

thehesiod commented on setup.py:
```diff
@@ -16,31 +17,29 @@
-        if line.startswith('#define H5_VERSION'):
-            hdf5_version = line.split()[2]
+        if line.startswith('#define H5_VERS_INFO'):
+            hdf5_version = line.split('"')[1]
```
looks like another weirdness injected for pep'ifying :(  I'm now worried about this tool

jswhit commented:
Nope - I did that (changed H5_VERSION to H5_VERS_INFO), and line.split() to
line.split('"')).

On Tue, Jun 20, 2017 at 12:48 PM, Alexander Mohr <notifications@github.com>
wrote:

> *@thehesiod* commented on this pull request.
> ------------------------------
>
> In setup.py
> <https://github.com/Unidata/netcdf4-python/pull/679#discussion_r123063197>
> :
>
> >      hdf5_version = None
>      for line in f:
> -        if line.startswith('#define H5_VERSION'):
> -            hdf5_version = line.split()[2]
> +        if line.startswith('#define H5_VERS_INFO'):
> +            hdf5_version = line.split('"')[1]
>
> looks like another weirdness injected for pep'ifying :( I'm now worried
> about this tool
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Unidata/netcdf4-python/pull/679#pullrequestreview-45232387>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAjYCZqTqZdYI8gEJYnXjoVUEItMeFulks5sGBQbgaJpZM4N-iUZ>
> .
>



-- 
Jeffrey S. Whitaker
NOAA/OAR/PSD  R/PSD1
325 Broadway, Boulder, CO, 80305-3328
Phone: (303)497-6313
FAX: (303)497-6449

thehesiod commented:
sounds good to me at least, less complexity the better :)
